src/state/user/hooks.tsx

```diff
@@ -139,7 +139,23 @@ const selectSelectedChartOption = createSelector(
 )
 
 export function useSelectedChartOption() {
-  return useSelector(selectSelectedChartOption)
+  return useSelector<AppState, AppState['user']['selectedChartOption']>(selectSelectedChartOption)
+}
+
+export function useUpdateSelectedChartOption(): [ChartOption | undefined, (selectedChartOption: ChartOption) => void] {
+  const dispatch = useDispatch<AppDispatch>()
```

Comment:
I guess`dispatch` can be declared outside of the hook as it will always hold the same reference. I've seen it repeated in many hooks...

Reply:
In think it's fine both ways, as long as we stick with 1 way and don't use both in our codebase.